Marlin/Marlin_main.cpp

```diff
@@ -4952,57 +4956,89 @@ inline void gcode_M226() {
+        available_output_pressure = 0;
+      }
+      // Find lowest pressure available
+      else if (current_tank_target < current_tank) {
+        available_output_pressure = (current_tank_target - PNEUMATIC_HYSTERESIS_PSI);
+      }
+    }
+    // Desired pressure value given
     if(code_seen('S')) {
       float psi = code_value();
-      // Desired pressure is more than maximum allowed output pressure
-      if(psi > OUTPUT_PSI_MAX) {
-        SERIAL_PROTOCOLPGM("WARNING: Desired Pressure Above Max Allowed Pressure (");
-        SERIAL_PROTOCOL(OUTPUT_PSI_MAX);
-        SERIAL_PROTOCOLPGM(" psi)");
-      }
-      // Desired pressure is less than minimum allowed output pressure
-      else if(psi < OUTPUT_PSI_MIN) {
-        SERIAL_PROTOCOLPGM("WARNING: Desired Pressure Below Min Allowed Pressure (");
+      // Desired pressure outside allowed range?
+      if((psi > OUTPUT_PSI_MAX) || (psi < OUTPUT_PSI_MIN)) {
+        SERIAL_PROTOCOLPGM("ERROR: Desired Pressure Outside Allowed Pressure Range (");
         SERIAL_PROTOCOL(OUTPUT_PSI_MIN);
+        SERIAL_PROTOCOLPGM(" - ");
+        SERIAL_PROTOCOL(OUTPUT_PSI_MAX);
         SERIAL_PROTOCOLPGM(" psi)");
       }
       // Desired pressure is available
-      else if((psi <= (current_tank - PNEUMATIC_HYSTERESIS_PSI))  && (psi <= (current_tank_target - PNEUMATIC_HYSTERESIS_PSI))) {
-        setOutputPressure(psi);
+      else if(psi <= available_output_pressure) {
+        regulator_setpoint = psi;
+        setOutputPressure(regulator_setpoint);
       }
       // Tank pressure is near zero, can set output to near zero
-      else if(((psi == 0) && (current_tank <= REGULATOR_LOW_P)) || ((psi == 0) && (current_tank_target <= REGULATOR_LOW_P))) {     
-        setOutputPressure(psi);
+      else if( (psi == 0) && !(house_air) ) {
+        if( (current_tank <= REGULATOR_LOW_P) || (current_tank_target <= REGULATOR_LOW_P) ) {
+          regulator_setpoint = psi;
+          setOutputPressure(regulator_setpoint);
+        }
       }
-      // Desired pressure not available
+      // Desired pressure NOT available
       else {
-        uint16_t available_output_pressure = 0;
-        
-        if ((current_tank <= REGULATOR_LOW_P) || (current_tank_target <= REGULATOR_LOW_P)) {
-          available_output_pressure = 0;
-        }
-        else if (current_tank <= current_tank_target) {
-          available_output_pressure = (current_tank - PNEUMATIC_HYSTERESIS_PSI);
-        }
-        else if (current_tank_target < current_tank) {
-          available_output_pressure = (current_tank_target - PNEUMATIC_HYSTERESIS_PSI);
-        }
-        SERIAL_PROTOCOLLNPGM("WARNING: Cannot output desired pressure due to insufficient tank pressure");
+        SERIAL_PROTOCOLLNPGM("ERROR: Insufficient tank pressure");
         SERIAL_PROTOCOLPGM("Available Tank Pressure: ");
         SERIAL_PROTOCOL(available_output_pressure);
         SERIAL_PROTOCOLPGM(" psi");
       }
-      SERIAL_EOL;
```

Comment:
@kevingelion Because I added a verbose output, had to move it down. It now resides on line 5041